Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/IncidentConsumer/IncidentConsumer.cs

```diff
@@ -50,18 +50,23 @@ internal async Task<string> GetAccessTokenAsync(string uri, ILogger log)
         {
             try
             {
-                var credential = new ClientCredential(ClientId, ClientKey);
-                var authenticationContext = new AuthenticationContext($"https://login.microsoftonline.com/{TenantId}");
-                var result = await authenticationContext.AcquireTokenAsync(uri, credential);
-                return result.AccessToken;
+                string authority = $"https://login.microsoftonline.com/{TenantId}";
+                IConfidentialClientApplication app = ConfidentialClientApplicationBuilder.Create(ClientId)
+                                                     .WithClientSecret(ClientKey)
+                                                     .WithAuthority(authority)
+                                                     .Build();
+                var authResult = await app.AcquireTokenForClient(
+                                     new[] { uri })
+                                 .ExecuteAsync()
+                                 .ConfigureAwait(false);
+                return authResult.AccessToken;
             }
             catch (Exception ex)
             {
                 log.LogError("GetAccessTokenAsync uri --> " + uri);
                 log.LogError("GetAccessTokenAsync ex --> " + ex.Message);
+                return null;
             }
-            throw new Exception();
```

Comment:
This line is not needed.  The function will either exit on ln 62 or ln 68. This exception re-throwing will never happen.